framework/helpers/BaseJson.php

```diff
@@ -22,6 +22,13 @@
  */
 class BaseJson
 {
+    /**
+     * @var bool|null Enables human readable output a.k.a. Pretty Print.
+     * This can useful for debugging during development but is not recommended in a production environment!
```

Comment:
This sentence should be removed IMHO. JSON pretty print is not only used when debugging, and we shouldn't discourage people from using it.

Reply:
As described here https://stackoverflow.com/a/20781384
> The extra white space that is inserted has negative effects for performance (generation, parsing, and transmission time).

Since this setting operates globally I think a warning is in place. Perhaps we can add that if Pretty Print is required for a specific use case it can be set via the `Json::encode()` `$options` parameter.

Reply:
Good point.
Ok to leave this as is, and add your reply in the [Performance section](https://github.com/yiisoft/yii2/blob/master/docs/guide/tutorial-performance-tuning.md)

@samdark ok for you?

Reply:
Fine to keep it as is,